src/Diagnostics.DataProviders/DataProviders/MdmDataProvider.cs

```diff
@@ -0,0 +1,185 @@
+        }
+
+        /// <summary>
+        /// Gets the list of dimension names for the metricId.
+        /// </summary>
+        /// <param name="metricNamespace">Metric namespace</param>
+        /// <param name="metricName">Metric name</param>
+        /// <returns>The list of dimension names for the metricId.</returns>
+        public async Task<IEnumerable<string>> GetDimensionNamesAsync(string metricNamespace, string metricName)
+        {
+            return await _mdmClient.GetDimensionNamesAsync(_configuration.MonitoringAccount, metricNamespace, metricName);
+        }
+
+        /// <summary>
+        /// Gets the dimension values for dimensionName satifying the dimensionFilters and
+        /// </summary>
+        /// <param name="metricNamespace">Metric namespace</param>
+        /// <param name="metricName">Metric name</param>
+        /// <param name="filter">The dimension filters representing the pre-aggregate dimensions. Create an emtpy include filter for dimension with no filter values. Requested dimension should also be part of this and should be empty.</param>
+        /// <param name="dimensionName">Name of the dimension for which values are requested.</param>
+        /// <param name="startTimeUtc">Start time for evaluating dimension values.</param>
+        /// <param name="endTimeUtc">End time for evaluating dimension values.</param>
+        /// <returns>Dimension values for dimensionName.</returns>
+        public async Task<IEnumerable<string>> GetDimensionValuesAsync(string metricNamespace, string metricName, List<Tuple<string, IEnumerable<string>>> filter, string dimensionName, DateTime startTimeUtc, DateTime endTimeUtc)
+        {
+            var contain = false;
+            foreach (var f in filter)
+            {
+                if(f.Item1.Equals(dimensionName, StringComparison.OrdinalIgnoreCase))
+                {
+                    contain = true;
+                    break;
+                }
+            }
+
+            if (!contain)
+            {
+                filter.Add(Tuple.Create(dimensionName, (IEnumerable<string>)new string[] { }));
+            }
+
+            return await _mdmClient.GetDimensionValuesAsync(new MetricIdentifier(_configuration.MonitoringAccount, metricNamespace, metricName), filter, dimensionName, startTimeUtc, endTimeUtc);
+        }
+
+        /// <summary>
+        /// Gets the time series.
+        /// </summary>
+        /// <param name="startTimeUtc">The start time UTC.</param>
+        /// <param name="endTimeUtc">The end time UTC.</param>
+        /// <param name="sampling">The sampling type.</param>
+        /// <param name="definition">The time series definition.</param>
+        /// <returns>The time series for the given definition.</returns>
+        public async Task<IEnumerable<DataTable>> GetTimeSeriesAsync(DateTime startTimeUtc, DateTime endTimeUtc, Sampling sampling, Tuple<string, string, IEnumerable<KeyValuePair<string, string>>> definition)
```

Comment:
I think detector writers will find it more of a hard time to create the tuple definition for `Tuple<string, string, IEnumerable<KeyValuePair<string, string>>> definition`. Instead consider creating separate parameters for the namespace, metricname, and the combination of mdm dimensions.